pkg/crypto/crypto.go

```diff
@@ -644,15 +644,15 @@ func MakeSelfSignedCAConfigForSubject(subject pkix.Name, expireDays int) (*TLSCe
 	}
 
 	caLifetime := time.Duration(caLifetimeInDays) * 24 * time.Hour
-	return makeSelfSignedCAConfigForSubjectAndDuration(subject, caLifetime)
+	return MakeSelfSignedCAConfigForSubjectAndDuration(subject, caLifetime)
 }
 
 func MakeSelfSignedCAConfigForDuration(name string, caLifetime time.Duration) (*TLSCertificateConfig, error) {
 	subject := pkix.Name{CommonName: name}
-	return makeSelfSignedCAConfigForSubjectAndDuration(subject, caLifetime)
+	return MakeSelfSignedCAConfigForSubjectAndDuration(subject, caLifetime)
 }
 
-func makeSelfSignedCAConfigForSubjectAndDuration(subject pkix.Name, caLifetime time.Duration) (*TLSCertificateConfig, error) {
+func MakeSelfSignedCAConfigForSubjectAndDuration(subject pkix.Name, caLifetime time.Duration) (*TLSCertificateConfig, error) {
```

Comment:
it's not clear why this should be exposed instead of the already exposed `MakeSelfSignedCAConfigForSubject` which has some checking and then delegates here.

Reply:
I think it makes sense to enable callers to provide the lifetime as a duration constant rather than a magic number of days, but that's just me. If you feel strongly I can remove this change.

Reply:
> I think it makes sense to enable callers to provide the lifetime as a duration constant rather than a magic number of days, but that's just me. If you feel strongly I can remove this change.

I'd like to remove this one.

Reply:
Done